vpr/SRC/base/SetupVPR.c

```diff
@@ -57,6 +57,7 @@ void SetupVPR(INP t_options *Options, INP bool TimingEnabled,
 		t_power_opts * PowerOpts) {
 	int i, j, len;
 
+	if (!Options->CircuitName) vpr_throw(VPR_ERROR_BLIF_F,__FILE__, __LINE__, "No blif file found in arguments (did you specify an architecture file?)\n");

```

Comment:
This extra error check is really separate from the connection base routing changes.